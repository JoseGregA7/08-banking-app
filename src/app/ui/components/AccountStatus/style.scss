.AccountStatus__wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
   width: 100%;
}

.AccountStatus__main {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 4px;
    box-shadow: 1px 1px rgba(241, 242, 245, 0.3), 2px 2px 15px rgba(223, 225, 236, 0.6);
    border-radius: 5px;
    width: fit-content;
}

.AccountStatus__title {
    font-size: 1rem;
    padding: 0;
    margin: 0;
}

.AccountStatus__content {
    font-size: 0.8rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    padding: 0;
    margin: 0;
}

.AccountStatus__list {
    list-style: none;
    padding: 0;
    margin: 0;
}

.AccountStatus__item {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: start;
    padding: 2px;
    margin: 0;
    list-style: none;

    strong {
        color: #666;
        font-weight: bold;
        margin-right: 5px;
    }
}