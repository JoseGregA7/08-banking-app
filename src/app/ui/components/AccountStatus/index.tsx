import { LABEL_MAPPING } from '../../../core/constants/account';
import { AccountStatusProps } from '../../../core/interfaces/state';
import './style.scss';

const AccountStatus = ({ accountInfo }: AccountStatusProps) => {
    const filteredData = accountInfo
        ? Object.entries(accountInfo)
            .filter(([key, value]) => value !== null)
            .map(([key, value]) => {
                const label = LABEL_MAPPING[key as keyof typeof LABEL_MAPPING] || key;
                return [label, value];
            })
        : [];

    return (
        <div
            className="AccountStatus__wrapper"
            role="region"
            aria-labelledby="account-status-title"
            data-testid="account-status-wrapper"
        >
            <div className="AccountStatus__main">
                <h2
                    id="account-status-title"
                    className="AccountStatus__title"
                    title="Estado de la Cuenta"
                    aria-label="Estado de la Cuenta"
                    data-testid="account-status-title"
                >
                    Estado de la Cuenta
                </h2>
                <div className="AccountStatus__content">
                    {filteredData.length === 0 ? (
                        <div
                            role="alert"
                            aria-live="assertive"
                            data-testid="account-status-no-data"
                        >
                            No hay datos disponibles.
                        </div>
                    ) : (
                        <ul className="AccountStatus__list" role="list" aria-labelledby="account-status-list">
                            {filteredData.map(([label, value]) => (
                                <li className="AccountStatus__item" key={label} role="listitem">
                                    <strong>{label}:</strong> {value}
                                </li>
                            ))}
                        </ul>
                    )}
                </div>
            </div>
        </div>
    );
};

export default AccountStatus;
