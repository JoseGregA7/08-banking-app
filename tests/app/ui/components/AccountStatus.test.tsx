import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { describe, expect, test, vitest } from 'vitest';
import AccountStatus from '../../../../src/app/ui/components/AccountStatus';

describe('AccountStatus', () => {
  const accountInfo = {
    id: 12,
    name: 'John Doe',
    amount: 4000,
  };

  test('should render account status details when accountInfo is provided', () => {
    const { getByTestId, getByText } = render(
      <AccountStatus accountInfo={accountInfo} />
    );
    expect(getByTestId('account-status-wrapper')).toBeInTheDocument();
    expect(getByTestId('account-status-title')).toHaveTextContent('Estado de la Cuenta');
    expect(getByText('12')).toBeInTheDocument();
    expect(getByText('Saldo Actual:')).toBeInTheDocument();
    expect(getByText('4000')).toBeInTheDocument();
    expect(getByText('John Doe')).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const { asFragment } = render(
      <AccountStatus accountInfo={accountInfo} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

});
