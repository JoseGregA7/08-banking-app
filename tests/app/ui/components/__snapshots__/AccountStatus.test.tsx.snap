// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AccountStatus > should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="account-status-title"
    class="AccountStatus__wrapper"
    data-testid="account-status-wrapper"
    role="region"
  >
    <div
      class="AccountStatus__main"
    >
      <h2
        aria-label="Estado de la Cuenta"
        class="AccountStatus__title"
        data-testid="account-status-title"
        id="account-status-title"
        title="Estado de la Cuenta"
      >
        Estado de la Cuenta
      </h2>
      <div
        class="AccountStatus__content"
      >
        <ul
          aria-labelledby="account-status-list"
          class="AccountStatus__list"
          role="list"
        >
          <li
            class="AccountStatus__item"
            role="listitem"
          >
            <strong>
              id:
            </strong>
             12
          </li>
          <li
            class="AccountStatus__item"
            role="listitem"
          >
            <strong>
              name:
            </strong>
             John Doe
          </li>
          <li
            class="AccountStatus__item"
            role="listitem"
          >
            <strong>
              Saldo Actual:
            </strong>
             4000
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
